### PR TITLE
Add scalar functions and tests for Float

### PR DESCRIPTION
**Pull request by akshatnair**
Testing: unit tests

### CODE REVIEW

**Comment by akshatnair** [comment]
@maciejgrzybek @cberner

**Comment by maciejgrzybek** [comment]
Link to internal Teradata review: https://github.com/Teradata/presto/pull/236

**Comment by akshatnair** [comment]
I will check why Travis build is failing

**Comment by akshatnair** [comment]
Travis is failing on master with the same errors on hive tests

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java [hunk 1]
```diff
@@ -448,6 +476,14 @@ public void testRound()
         assertFunction("round(-3.5)", DOUBLE, -4.0);
         assertFunction("round(-3.5001)", DOUBLE, -4.0);
         assertFunction("round(-3.99)", DOUBLE, -4.0);
+        assertFunction("round(FLOAT '3.0')", FLOAT, 3.0f);
```
Comment (cberner) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java]: Looks like these are duplicated below

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java [hunk 2]
```diff
@@ -481,6 +517,14 @@ public void testRound()
         assertFunction("round(TINYINT '3', 1)", TINYINT, (byte) 3);
         assertFunction("round(SMALLINT '3', SMALLINT '1')", SMALLINT, (short) 3);
         assertFunction("round(SMALLINT '3', 1)", SMALLINT, (short) 3);
+        assertFunction("round(FLOAT '3.0', 0)", FLOAT, 3.0f);
```
Comment (cberner) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java]: These seem to be duplicated below

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestColorFunctions.java [hunk 3]
```diff
@@ -130,6 +131,7 @@ public void testRenderDouble()
             throws Exception
     {
         assertEquals(render(1234.5678, color(toSlice("red"))), toSlice("\u001b[38;5;1m1234.5678\u001b[0m"));
+        assertEquals(render(1234.5678f, color(toSlice("red"))), toSlice(format("\u001b[38;5;1m%s\u001b[0m", (double) 1234.5678f)));
```
Comment (cberner) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestColorFunctions.java]: Are these relevant to the float functions you added? If they're just more tests for the color type, add them in a separate commit

**Comment by cberner** [comment]
Looks good except for some minor comments

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestColorFunctions.java (hunk 3 above)
Comment (akshatnair) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestColorFunctions.java]: No I just wanted to test color functions with floats, I can add them in a different commit

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java (hunk 2 above)
Comment (akshatnair) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java]: It is testing 3 different scenarios, there is round(3.0) round(3.0, 0) and then round(3.0, 1)

**Comment by akshatnair** [comment]
@cberner comments have been addressed

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java (hunk 2 above)
Comment (cberner) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java]: Oh ya, missed that

**Comment by cberner** [comment]
Merged. Thanks!
